plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/SchemaProperty.kt

```diff
@@ -0,0 +1,62 @@
+package org.jetbrains.kotlinx.dataframe.plugin
+
+import org.jetbrains.kotlin.fir.expressions.FirAnnotation
+import org.jetbrains.kotlin.fir.expressions.builder.buildAnnotationArgumentMapping
+import org.jetbrains.kotlin.fir.expressions.builder.buildLiteralExpression
+import org.jetbrains.kotlin.fir.resolve.defaultType
+import org.jetbrains.kotlin.fir.types.ConeKotlinType
+import org.jetbrains.kotlin.fir.types.ConeTypeProjection
+import org.jetbrains.kotlin.fir.types.builder.buildResolvedTypeRef
+import org.jetbrains.kotlin.name.Name
+import org.jetbrains.kotlin.types.ConstantValueKind
+import org.jetbrains.kotlinx.dataframe.codeGen.ValidFieldName
+import org.jetbrains.kotlinx.dataframe.plugin.utils.Names
+
+data class SchemaProperty(
+    val marker: ConeTypeProjection,
+    val propertyName: PropertyName,
+    val dataRowReturnType: ConeKotlinType,
+    val columnContainerReturnType: ConeKotlinType,
+    val override: Boolean = false
+)
+
+data class PropertyName(val identifier: Name, val columnNameAnnotation: FirAnnotation?) {
```

Comment:
might deserve its own file? Makes it easier to find, especially since it's used very often

Reply:
Moved it under extensions.impl package